src/treelearner/ocl/histogram16.cl

```diff
@@ -642,22 +558,14 @@ R""()
         ushort bank_id = (i + offset) & BANK_MASK;
         stat_val += gh_hist[bin_id * HG_BIN_MULT + bank_id * 2 * DWORD_FEATURES + is_hessian_first * DWORD_FEATURES + feature_id];
     }
-    if (ltid < LOCAL_SIZE_0 / 2) {
-        // first 128 threads accumulate the 8 * 16 = 128 counter values
-        bin_id = ltid >> LOG2_DWORD_FEATURES; // bits 3 - 6 range 0 - 16 is bin ID
-        offset = (ltid >> LOG2_DWORD_FEATURES) & BANK_MASK; // helps avoid LDS bank conflicts
-        for (int i = 0; i < NUM_BANKS; ++i) {
-            ushort bank_id = (i + offset) & BANK_MASK;
-            cnt_val += cnt_hist[bin_id * CNT_BIN_MULT + bank_id * DWORD_FEATURES + feature_id];
-        }
-    }
     
     // now thread 0 - 7  holds feature 0 - 7's gradient for bin 0 and counter bin 0
     // now thread 8 - 15 holds feature 0 - 7's hessian  for bin 0 and counter bin 1
     // now thread 16- 23 holds feature 0 - 7's gradient for bin 1 and counter bin 2
     // now thread 24- 31 holds feature 0 - 7's hessian  for bin 1 and counter bin 3
     // etc,
 
+    // FIXME: correct way to fix hessians
 #if CONST_HESSIAN == 1
```

Comment:
@huanzhang12 could you help to fix the hessian fixing here?

Reply:
Can you elaborate a little bit on what you want to do here? Do we need to add anything new to the histogram?

Reply:
Thanks! As the count is removed, the naive solution is to add gh_hist as before and remove the cnt_hist. However, for const_hessian cases, we can use `+1` in h_hist firstly and then multiply all h_hist with hessians[0], If `+1` is faster than `+hessians[i]`. Therefore, here we need to multiply all h_hist with hessians[0].